InnerEye/ML/reports/classification_report.py

```diff
@@ -52,84 +54,220 @@ class ReportedMetrics(Enum):
+    else:
+        raise ValueError(f"Plot type must be either 'pr' or 'roc' (received '{plot_type}')")
     ax.set_title(title)
+    ax.set_xlabel(xlabel)
+    ax.set_ylabel(ylabel)
+    ax.set_xlim(0, 1)
+    ax.set_ylim(0, 1)
+    ax.grid(lw=1, color='lightgray')
 
-    if print_coords:
-        # write values of points
-        for x, y in zip(x_values, y_values):
-            ax.annotate(f"{x:0.3f}, {y:0.3f}", xy=(x, y), xytext=(15, 0), textcoords='offset points')
 
-
-def plot_pr_and_roc_curves(labels_and_model_outputs: LabelsAndPredictions) -> None:
+def plot_pr_and_roc_curves(labels_and_model_outputs: LabelsAndPredictions, axs: Optional[Sequence[Axes]] = None,
+                           plot_kwargs: Optional[Dict[str, Any]] = None) -> None:
     """
-    Given a LabelsAndPredictions object, plot the ROC and PR curves.
+    Given labels and model outputs, plot the ROC and PR curves.
+    :param labels_and_model_outputs:
+    :param axs: Pair of axes objects onto which to plot the ROC and PR curves, respectively. New axes are created by
+    default.
+    :param plot_kwargs: Plotting options to be passed to both `ax.plot(...)` calls.
     """
-    print_header("ROC and PR curves", level=3)
-    _, ax = plt.subplots(1, 2)
+    if axs is None:
+        _, axs = plt.subplots(1, 2)
+    if plot_kwargs is None:
+        plot_kwargs = {}
 
     fpr, tpr, thresholds = roc_curve(labels_and_model_outputs.labels, labels_and_model_outputs.model_outputs)
+    axs[0].plot(fpr, tpr, **plot_kwargs)
+    format_pr_or_roc_axes('roc', axs[0])
 
-    plot_auc(fpr, tpr, "ROC Curve", ax[0])
     precision, recall, thresholds = precision_recall_curve(labels_and_model_outputs.labels,
                                                            labels_and_model_outputs.model_outputs)
-    plot_auc(recall, precision, "PR Curve", ax[1])
+    axs[1].plot(recall, precision, **plot_kwargs)
+    format_pr_or_roc_axes('pr', axs[1])
+
+    plt.show()
+
+
+def plot_scores_and_summary(all_labels_and_model_outputs: Sequence[LabelsAndPredictions],
+                            scoring_fn: Callable[[LabelsAndPredictions], Tuple[np.ndarray, np.ndarray]],
+                            confidence_interval_width: float = .8,
+                            ax: Optional[Axes] = None) -> Tuple[List, Any]:
+    """
+    Plot a collection of score curves along with the (vertical) median and confidence interval (CI).
+
+    Each plotted curve is interpolated onto a common horizontal grid, and the median and CI are computed vertically
+    at each horizontal location.
+    :param all_labels_and_model_outputs: Collection of ground-truth labels and model predictions (e.g. for various
+    cross-validation runs).
+    :param scoring_fn: A scoring function mapping a `LabelsAndPredictions` object to X and Y coordinates for plotting.
+    :param confidence_interval_width: A value in [0, 1] representing what fraction of the data should be contained in
```

Comment:
To me it's weird to call that a confidence interval. If anything CI at 95% should be ~[mean +/- 2*std]

Reply:
I disagree 🙂 strictly speaking, an X% CI is any interval that contains X% of the probability mass. it just happens that 95% is _mean +/- 1.96*std_ for a normal distribution. for these ROC/PR curves I don't think a normal assumption is appropriate (e.g. it's bounded in [0, 1]), so I went with quantiles.

Reply:
I'll just rename it 'interval' to avoid confusion

Reply:
Yes but here you have a sample (of size n=number_splits) of the distribution over estimators not the true distribution so strictly speaking it's not a CI . But, Yes 1.96 assumues normal i agree not always respected for sure.